src/components/PostDetails.tsx

```diff
@@ -66,26 +38,19 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(commentsAction.setComments(newComment));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(commentsAction.setError);
```

Comment:
you can also create async Thunk for it
And handle "fulfilled" and "rejected" in extra reducers